graylog2-server/src/main/java/org/graylog2/tokenusage/PaginatedTokenUsageService.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.tokenusage;
+
+import com.mongodb.client.MongoCollection;
+import jakarta.inject.Inject;
+import org.graylog2.database.MongoCollections;
+import org.graylog2.database.PaginatedList;
+import org.graylog2.database.pagination.MongoPaginationHelper;
+import org.graylog2.rest.models.SortOrder;
+import org.graylog2.rest.models.tokenusage.TokenUsageDTO;
+import org.graylog2.search.SearchQuery;
+import org.graylog2.security.AccessTokenImpl;
+
+public class PaginatedTokenUsageService {
```

Comment:
Not sure if we want to have a paginated service per use case. Ideally, we would have one service responsible for the data layer. AccessTokenServiceImpl is a bit old, we could refactor it to the new approach and add the pagination method there. what do you think? @fpetersen-gl  @moesterheld

Reply:
Just talked with Florian makes sense we, don't want to expose the additional information